src/Cache.ts

```diff
@@ -272,10 +275,10 @@ export class Cache {
                 ) {
                     // We went past the current section (or this is the first task).
                     // Find the section that is relevant for this task and the following of the same section.
-                    currentSection = Cache.getSection({
-                        lineNumberTask: listItem.position.start.line,
-                        sections: fileCache.sections,
-                    });
+                    currentSection = Cache.getSection(
+                        listItem.position.start.line,
+                        fileCache.sections,
+                    );
```

Comment:
I am comfortable with the changing to this style of arguments. I have long wondered what the benefit is of the original style.

Personally, I would have made one commit that changed the argument styles, and then another commit that changed the behaviour. This makes the reviewing easier, although I appreciate that each commit in this project takes 5-10 seconds because of the amount of analysis and checks going on.

Thinking out loud: no action required: In the longer term, we might look at updating the arguments systematically, rather than piecemeal... Although I am conscious that doing this would make it hard to merge code from sytone's tasks-x fork.

Reply:
Thank you. That really helps.

However, for the avoidance of doubt, I would very much prefer this style not to be applied systematically to either tasks or tasks-x for the foreseeable future, please.

I am in the happy position of struggling to keep up with both the rate of very welcome contributions to tasks and my own design-based refactoring that will simplify the code a lot, and allow me to fix one or two really quite serious errors, that are just not making it to the top of my list.

Thanks.